pkg/store/bucket.go

```diff
@@ -962,8 +962,24 @@ func (b *blockSeriesClient) MergeStats(stats *queryStats) *queryStats {
 	return stats
 }
 
+type sortedMatchers []*labels.Matcher
+
+func newSortedMatchers(matchers []*labels.Matcher) sortedMatchers {
+	sort.Slice(matchers, func(i, j int) bool {
+		if matchers[i].Type == matchers[j].Type {
+			if matchers[i].Name == matchers[j].Name {
+				return matchers[i].Value < matchers[j].Value
+			}
+			return matchers[i].Name < matchers[j].Name
+		}
+		return matchers[i].Type < matchers[j].Type
+	})
+
+	return matchers
```

Comment:
nit: could reduce nesting by:


```go
sort.Slice(matchers, func(i, j int) bool {
   if matchers[i].Type != matchers[j].Type {
      return matchers[i].Type < matchers[j].Type
   }
			
   if matchers[i].Name == matchers[j].Name {
      return matchers[i].Value < matchers[j].Value
   }

   return matchers[i].Name < matchers[j].Name
})
```

Reply:
In the original PR I copied/pasted the same logic https://github.com/thanos-io/thanos/pull/6575 so I'd prefer to keep the same to reduce discrepancy :smile:

Reply:
You could update both PR's 😏 😄